src/jit_no_ir.h

```diff
@@ -149,64 +163,34 @@ class EmitX64 {
         return reinterpret_cast<EmitX64*>(this_ptr)->LookupNewBlock();
     }
 
-    int count = 0;
-
     BlockFunc LookupNewBlock() {
         if (cycles_remaining <= 0) {
             return nullptr;
         }
 
-               // Lookup and compile next block
+        // State for bank exchange.
         blk_key.pc = regs.pc;
-
-               // State for bank exchange.
-        blk_key.cfgi.raw = regs.cfgi.raw & Cfg::Mask();
-        blk_key.cfgj.raw = regs.cfgj.raw & Cfg::Mask();
-        blk_key.cfgib.raw = regs.cfgib.raw & Cfg::Mask();
-        blk_key.cfgjb.raw = regs.cfgjb.raw & Cfg::Mask();
-        blk_key.stepi0 = regs.stepi0;
-        blk_key.stepj0 = regs.stepj0;
-        blk_key.stepi0b = regs.stepi0b;
-        blk_key.stepj0b = regs.stepj0b;
-
-               // Current state
-        blk_key.curr.mod0.raw = regs.mod0.raw & Mod0::Mask();
-        blk_key.curr.mod1.raw = regs.mod1.raw & Mod1::Mask();
-        blk_key.curr.mod2.raw = regs.mod2.raw;
-        blk_key.curr.ar = regs.ar;
-        for (size_t i = 0; i < blk_key.curr.ar.size(); i++) {
-            blk_key.curr.ar[i].raw &= ArU::Mask();
-        }
-        blk_key.curr.arp = regs.arp;
-        for (size_t i = 0; i < blk_key.curr.arp.size(); i++) {
-            blk_key.curr.arp[i].raw &= ArpU::Mask();
-        }
-
-               // Shadow state.
-        blk_key.shadow.mod0.raw = regs.mod0b.raw & Mod0::Mask();
-        blk_key.shadow.mod1.raw = regs.mod1b.raw & Mod1::Mask();
-        blk_key.shadow.mod2 = regs.mod2b;
-        blk_key.shadow.ar = regs.arb;
-        for (size_t i = 0; i < blk_key.shadow.ar.size(); i++) {
-            blk_key.shadow.ar[i].raw &= ArU::Mask();
-        }
-        blk_key.shadow.arp = regs.arpb;
-        for (size_t i = 0; i < blk_key.shadow.arp.size(); i++) {
-            blk_key.shadow.arp[i].raw &= ArpU::Mask();
-        }
+        std::memcpy(&blk_key.cfgi, &regs.cfgi, sizeof(u16) * 8);
```

Comment:
If memcpy'ing directly then it's very highly recommended to at least put a `static_assert` in BlockKey to verify there's no padding anywhere. Maybe even static_assert'ing the offsetof of each member, but should be safe to skip that if lazy